Create integrate-property-editors.md (#6229)
umbraco/UmbracoDocs: 14/umbraco-cms/extending/property-editors/integrate-property-editors.md

# Integrate Property Editors in your UI

> Property Editors can be used in any places of Backoffice and you can implement them too.

Reviewer: ```suggestion
Property Editors can be used and implemented anywhere in the Umbraco Backoffice.
```

## Property & Property Dataset Components

> To integrate one or more Property Editors, it can be done by using two Components.
> The Property Dataset component and a Property component.

Reviewer: ```suggestion
Integrating one or more Property Editors requires two Components: the Property Dataset component and a Property component.
```

Author: Well, it can be done in other ways. I understand its not so clear.. The point here is that this is the simplest way to implement it.

Author: .. and at this point, the only option we will be describing in our Docs. The other one is rather advanced.

Reviewer: Aha! Could this be an option instead then:
"The simplest way to integrate one or more Property Editors is done using two Components: : the Property Dataset component and a Property component."

Author: Yes, rigtig godt :-)

> The `umb-property` component renders a property using a Property Editor UI.
> The `umb-property-dataset` component provides the dataset for any properties within, This holds the data and keeps it presented no matter if an actual property is rendered. Making it possible to hide properties in tabs or other ways.

Reviewer: ```suggestion
The `umb-property` component renders a property using a Property Editor UI.

The `umb-property-dataset` component provides the dataset for any properties within. It holds the data even if the actual property is not rendered. This makes it possible to hide properties in tabs or other ways.
```

In the following example a dataset is implemented with two properties:

```
<umb-property-dataset .value=${this.data} @change=${this.#onDataChange}>
    <umb-property
        label="Textual input"
        description="Example of text editor"
        alias="textProperty"
        property-editor-ui-alias="Umb.PropertyEditorUi.TextBox"></umb-property>
    <umb-property
        label="List of options"
        description="Example of dropdown editor"
        alias="listProperty"
        .config=${[
            {
                alias: 'multiple',
                value: false,
            },
            {
                alias: 'items',
                value: ['First Option', 'Second Option', 'Third Option'],
            },
        ]}
        property-editor-ui-alias="Umb.PropertyEditorUi.Dropdown"></umb-property>
</umb-property-dataset>
```

Notice how the values of the properties are handled by the dataset, leaving you with one component to integrate.

> * [Read more about Property Dataset here](../property-dataset/README.md)

Reviewer: ```suggestion
[Read more about Property Dataset here](../property-dataset/README.md)
```
